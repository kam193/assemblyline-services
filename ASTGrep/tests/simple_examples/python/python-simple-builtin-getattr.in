# Different quoting styles
getattr(builtins, "exec")("print('hello')")
getattr(builtins, 'exec')("print('hello')")
getattr(builtins, b'exec')("print('hello')")
getattr(builtins, b"exec")("print('hello')")

getattr(__import__("builtins"), "exec")("print('hel\"lo')")
getattr(__import__('builtins'), 'exec')("print('hello')")
getattr(__import__(b'builtins'), b'exec')("print('hello')")
getattr(__import__(b"builtins"), b"exec")("print('hello')")
getattr(__import__(b"builtins"), sth("aaa"))("print('hello')")

getattr(__builtins__, "getattr")

# Example of false positives
saveFile = QAction(QIcon('pyfem/qt/img/save128.png'),'&Save', self )