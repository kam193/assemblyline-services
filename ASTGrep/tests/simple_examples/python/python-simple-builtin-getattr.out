
print('hello')
print('hello')
print('hello')
print('hello')

print('hel"lo')
print('hello')
print('hello')
print('hello')
getattr(builtins, sth("aaa"))("print('hello')")

getattr


saveFile = QAction(QIcon('pyfem/qt/img/save128.png'),'&Save', self )